R/template.R

```diff
@@ -77,6 +73,244 @@ metric_summarizer <- function(metric_nm,
                               case_weights = NULL,
                               ...,
                               metric_fn_options = list()) {
+  signal_soft_deprecated(
+    paste(
+      "`metric_summarizer()` has been soft-deprecated as of version 1.2.0.",
+      "Please see `?metric_summarizer()` for further instructions."
+      )
+  )
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+  case_weights <- enquo(case_weights)
+
+  validate_not_missing(truth, "truth")
+  validate_not_missing(estimate, "estimate")
+
+  # Explicit handling of length 1 character vectors as column names
+  nms <- colnames(data)
+  truth <- handle_chr_names(truth, nms)
+  estimate <- handle_chr_names(estimate, nms)
+
+  finalize_estimator_expr <- rlang::expr(
+    finalize_estimator(!! truth, estimator, metric_nm)
+  )
+
+  metric_tbl <- dplyr::summarise(
+    data,
+    .metric = metric_nm,
+    .estimator = eval_tidy(finalize_estimator_expr),
+    .estimate = metric_fn(
+      truth = !! truth,
+      estimate = !! estimate,
+      !!! spliceable_estimator(estimator),
+      na_rm = na_rm,
+      !!! spliceable_event_level(event_level),
+      !!! spliceable_case_weights(case_weights),
+      !!! metric_fn_options
+    )
+  )
+
+  dplyr::as_tibble(metric_tbl)
+}
+
+#' Developer function for summarizing new numeric metrics
```

Comment:
I think we should do better than old Davis and actually add some tests for these 3 functions

Reply:
done in https://github.com/tidymodels/yardstick/pull/322/commits/b0cc93c05c966ff6448186fd920bac3c73251f3f